tests/filecheck/dialects/complex/complex_canonicalize.mlir

```diff
@@ -0,0 +1,334 @@
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %exp = complex.exp %complex1 : complex<f32>
+  %log = complex.log %exp : complex<f32>
+  return %log : complex<f32>
+
+  // CHECK:       %[[COMPLEX:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[COMPLEX]] : complex<f32>  
+}
+
+
+// CHECK-LABEL: func @test_complex_exp_log
+func.func @test_complex_exp_log() -> complex<f32> {
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %log = complex.log %complex1 : complex<f32>
+  %exp = complex.exp %log : complex<f32>
+  return %exp : complex<f32>
+
+  // CHECK:       %[[COMPLEX:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[COMPLEX]] : complex<f32>   
+}
+
+
+// CHECK-LABEL: func @test_complex_conj_conj
+func.func @test_complex_conj_conj() -> complex<f32> {
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %conj1 = complex.conj %complex1 : complex<f32>
+  %conj2 = complex.conj %conj1 : complex<f32>
+  return %conj2 : complex<f32>
+
+  // CHECK:       %[[COMPLEX:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[COMPLEX]] : complex<f32>  
+}
+
+
+// CHECK-LABEL: func @test_complex_add_sub_lhs
+func.func @test_complex_add_sub_lhs() -> complex<f32> {
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %complex2 = complex.constant [0.0 : f32, 2.0 : f32] : complex<f32>
+  %sub = complex.sub %complex1, %complex2 : complex<f32>
+  %add = complex.add %sub, %complex2 : complex<f32>
+  return %add : complex<f32>
+
+  // CHECK:       %[[ADD:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[ADD]] : complex<f32>
+}
+
+
+// CHECK-LABEL: func @test_complex_add_sub_rhs
+func.func @test_complex_add_sub_rhs() -> complex<f32> {
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %complex2 = complex.constant [0.0 : f32, 2.0 : f32] : complex<f32>
+  %sub = complex.sub %complex1, %complex2 : complex<f32>
+  %add = complex.add %complex2, %sub : complex<f32>
+  return %add : complex<f32>
+
+  // CHECK:       %[[ADD:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[ADD]] : complex<f32>
+}
+
+
+// CHECK-LABEL: func @test_complex_sub_add_lhs
+func.func @test_complex_sub_add_lhs() -> complex<f32> {
+  %complex1 = complex.constant [1.0 : f32, 0.0 : f32] : complex<f32>
+  %complex2 = complex.constant [0.0 : f32, 2.0 : f32] : complex<f32>
+  %add = complex.add %complex1, %complex2 : complex<f32>
+  %sub = complex.sub %add, %complex2 : complex<f32>
+  return %sub : complex<f32>
+
+  // CHECK:       %[[SUB:.*]] = complex.constant [1.000000e+00 : f32, 0.000000e+00 : f32] : complex<f32>
+  // CHECK-NEXT:  return %[[SUB]] : complex<f32>
+}
```

Comment:
I'm not sure this test makes sense, it's just two steps of constant folding that we already check individually? What's the advantage of keeping this test?